### ISSUE TITLE
"Merges the description in the normal object" is modifying the object on the page

### ISSUE THREAD

**Issue by xPaw** [Nuklon/Steam-Economy-Enhancer#201]
This:

https://github.com/Nuklon/Steam-Economy-Enhancer/blob/8f90b9950c7ab482a73f7c52348545ff747db4eb/code.user.js#L2215-L2216

I was writing some code for steamdb extension, and assumed the properties I saw on the `item` were present, but I should have been looking at `item.description` instead.

Is this merging actually needed?

**Comment by Nuklon** [comment]
On inventory page, `m_rgAssets `holds only basic info where the bulk of the data is inside `description`. On the market page, `m_rgAssets `contains everything directly and there the `description `is empty. It's likely possible to remove this merging entirely, but then all uses need to be checked as the properties are likely under `description`.

**Comment by xPaw** [comment]
I've noticed some code already does this, like so:

https://github.com/Nuklon/Steam-Economy-Enhancer/blob/c5b059bd4cee6302a277dc293cec362f9ca5b1bc/code.user.js#L820-L823

**Comment by Nuklon** [comment]
Yeah, I think it's possible to disable merging, but I think all usages need to be well checked, I think the inventory stuff is pretty easy to fix as it's mostly under the `description`.

**Comment by nolddor** [comment]
@Nuklon maybe we can do something like...:
```
if (typeof value === 'object') {

    const obj = {};
    Object.assign(obj, value, value.description);
    // Includes the id of the inventory item.
    value['id'] = key;
    arr.push(obj);
}
```
To not modify steam objects that does not belong to us?
